#!/bin/bash -x
. novarc

SERIES="trusty xenial"

# Have to use the glance image URL directly because of a bug:
# See https://bugs.launchpad.net/ubuntu/+source/python-glanceclient/+bug/1579768

mkdir -p ~/images
for series in $SERIES;
do
	if [ ! -f ~/images/$series-server-cloudimg-amd64-disk1.img ];
	then
		wget -O ~/images/$series-server-cloudimg-amd64-disk1.img http://cloud-images.ubuntu.com/$series/current/$series-server-cloudimg-amd64-disk1.img
	fi
        glance image-create --name="${series}-kvm" --progress \
		--container-format=bare --disk-format=qcow2 \
		--property hypervisor-type=qemu \
		--file ~/images/$series-server-cloudimg-amd64-disk1.img

	if [ ! -f ~/images/$series-server-cloudimg-amd64-root.tar.gz ];
	then
		wget -O ~/images/$series-server-cloudimg-amd64-root.tar.gz http://cloud-images.ubuntu.com/$series/current/$series-server-cloudimg-amd64-root.tar.gz
        fi
        glance image-create --name="${series}-lxd" --progress \
		--container-format=bare --disk-format=root-tar \
		--property architecture="x86_64" \
		--property hypervisor-type=lxd \
		--file ~/images/$series-server-cloudimg-amd64-root.tar.gz
	
done
